mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -270,7 +275,9 @@ class GeneralizedLinearRegression @Since("2.0.0") (@Since("2.0.0") override val
           .setParent(this))
       val trainingSummary = new GeneralizedLinearRegressionTrainingSummary(dataset, model,
         wlsModel.diagInvAtWA.toArray, 1, getSolver)
-      return model.setSummary(trainingSummary)
+      model.setSummary(trainingSummary)
+      instr.logSuccess(model)
+      return model
```

Comment:
would you mind rewriting this if statement to:

````scala
val model = if (familyObj == Gaussian && linkObj == Identity) {
  ...
} else {
  ...
}
instr.logSuccess(model)
model
````

Then we can avoid the code duplication?